### PR TITLE
Make Kubernetes plugin API server configurable for hosted K8s clusters

### PR DESCRIPTION
**Pull request by riceo**
**What this PR does / why we need it**:

Some hosted Kubernetes clusters are generating certificates that are not signing the domain referenced in the Configmap generated by this chart for the Kubernetes Plugin API URL. This causes all jobs to fail with certificate errors.

This is the case for Oracle's Container Engine service , and appears to be the same for Azure's Container Engine service too. See:

* https://github.com/jenkins-x/jx/issues/603 
* https://github.com/jenkins-x/jx/issues/1184

Both issues are for Jenkins X, but I am seeing the same issue on Jenkins installed via this chart. 

This change makes the Kubernetes Plugin URL dynamic, but defaults to the value it was previously hardcoded to.

ping @lachie83 @viglesiasce

Signed-off-by: Aaron Rice <cube@sneakybox.com>

### CODE REVIEW

**Comment by cpanato** [comment]
@riceo there is a conflict a rebase should fix and also can you sign the DCO - https://www.helm.sh/blog/helm-dco/index.html ? thanks!

**Comment by riceo** [comment]
@cpanato Thanks, all sorted.

**Comment by cpanato** [comment]
@riceo I think all commits need to sign off, can you check that please? you can follow these steps https://github.com/helm/charts/pull/7461/checks?check_run_id=15026822

File: stable/jenkins/values.yaml [hunk 1]
```diff
@@ -125,6 +125,9 @@ Master:
     # - secretName: jenkins.cluster.local
     #   hosts:
     #     - jenkins.cluster.local
+  # Kubernetes API url. Some cloud providers, e.g Azure Container Service (AKS), Oracle Container Service (OKE), require this
+  # to be set to https://kubernetes.default.svc, as the default is not included on their cluster generated certs.
+  KubernetesServerUrl: https://kubernetes.default
```
Comment (paulczar) [inline stable/jenkins/values.yaml:130]: Can you add this to README.md ?

Comment (paulczar) [inline stable/jenkins/values.yaml:130]: would changing the default to be https://kubernetes.default.svc make it more compatible across the board?

**Pull request closed without merging**